### PR TITLE
This is the Todo List project for Microverse curriculum

### PR DESCRIPTION
**Pull request by carlosprietobarron**
The name of this project is "TasksList CandA".

It's an application that helps the user to keep a list of task to do grouped by subject or projects with an initial project of "General Task", that it's used to wrap unsorted tasks, you can add as many projects as you desire and as many task for project as you need.

It use Firebase as its database and also LocalStorage to keep the tasks of the user.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,43 @@
...
+
+## Acknowledgments
+
+- Hat tip to anyone whose code was used
+- Inspiration
+- etc
+
+## 📝 License
+
+This project is [MIT](lic.url) licensed.
```
Comment (lekegitrepo) [inline README.md]: - **_Add instructions on how to clone the project and run it locally._**

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      new Task(doc.id, doc.data().prjId, doc.data().name,
+        doc.data().notes, doc.data().priority, doc.data().dueDate),
+      // {
+      //   "id": doc.id,
+      //   "prjId": doc.data().prjId,
+      //   "name": doc.data().name,
+      //   "notes": doc.data().notes,
+      //   "priority": doc.data().priority,
+      //   "dueDate": doc.data().dueDate
+      // }
```
Comment (lekegitrepo) [inline src/index.js]: - **_Remove all commented codes from your code._**

File: src/project.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+import { Task } from './task';
+
+class Project {
+  constructor(id, name, description, dueDt, priority, progress, nextTask) {
+    this.id = id;
+    this.name = name;
+    this.description = description;
+    this.dueDt = dueDt;
+    this.priority = priority;
+    // this.progress = progress;
```
Comment (lekegitrepo) [inline src/project.js]: - **_Remove all commented codes from your code._**

File: src/projectTab.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  };
+
+  const createAddTab = () => {
+    const navtabs = document.getElementById('navtabs');
+    const anchor = document.createElement('a');
+    anchor.setAttribute('class', 'nav-link');
+    anchor.setAttribute('href', '#tab-content');
+    anchor.textContent = 'Add Project';
+    anchor.onclick = tabContent.toggleForm;
+    // anchor.prjId = `pr-${tabsCount}`;
```
Comment (lekegitrepo) [inline src/projectTab.js]: - **_Remove all commented codes from your code._**

File: src/tabContent.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  //     ulNav.appendChild(liTask);
+
+  //     const anchor = document.createElement('a');
+  //     domUtils.setAttributes(anchor, {
+  //       class: 'nav-link',
+  //       href: '#',
+  //       textContent: task.name,
+  //     });
+  //   });
+  // };
```
Comment (lekegitrepo) [inline src/tabContent.js]: - **_Remove all commented codes from your code._**

File: src/task.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+      notes: params.notes,
+      priority: params.priority,
+      dueDate: params.dueDate,
+    }).then(async () => {
+      if (state === 'new') {
+        project.tasks.push(new Task(params.tskId, params.prjId, params.name, params.notes, params.priority, params.dueDate));
+        project.tasks[project.tasks.length - 1].localStore();
+        taskPanel.deleteAddBtn(); // pendiente
+        const lastTsk = project.tasks[project.tasks.length - 1];
+        // taskPanel.createTaskBtn(lastTsk);
```
Comment (lekegitrepo) [inline src/task.js]: - **_Remove all commented codes from your code._**

File: src/task.js [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    db.collection('projects').doc(docId).collection('tasks').get()
+      .then(
+        (snaptshot) => {
+          snaptshot.docs.forEach((doc) => {
+            console.log(doc.data());
+            addTask(doc);
+          });
+        },
+      );
+  }; */
```
Comment (lekegitrepo) [inline src/task.js]: - **_Remove all commented codes from your code._**

File: src/taskPanel.js [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+    element.value = '';
+    const rbs = document.querySelectorAll('input[name="tskRadios"]');
+
+    rbs.forEach(rb => {
+      rb.checked = false;
+    });
+
+    // for (const rb of rbs) {
+    //   rb.checked = false;
+    // }
```
Comment (lekegitrepo) [inline src/taskPanel.js]: - **_Remove all commented codes from your code._**

File: src/taskPanel.js [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+    const anchorItem = document.createElement('a');
+    domUtils.setAttributes(anchorItem, {
+      id: 'tsk-link-id',
+      class: 'nav-link tsk-link',
+      href: '#',
+    });
+    anchorItem.textContent = 'New Task';
+    anchorItem.onclick = createTaskForm;
+    anchorItem.prjId = prjId;
+    // anchorItem.id = `btn-${tskId}`
```
Comment (lekegitrepo) [inline src/taskPanel.js]: - **_Remove all commented codes from your code._**

File: src/taskPanel.js [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+      }
+    }
+    const liItem = document.createElement('li');
+    domUtils.setAttributes(liItem, {
+      class: 'nav-item',
+    });
+    const prjId = project.id;
+    const idx = project.tasks.length;
+    const tskId = `${project.id}-tsk-${idx}`;
+    // let newIx = DBproject.getNextTaskId(prjId);
```
Comment (lekegitrepo) [inline src/taskPanel.js]: - **_Remove all commented codes from your code._**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Request changes :recycle:
### Hi @carlosprietobarron :clap: :clap:
#### You have done a good job :+1: However, you need to implement the following requirements before this project can be approved.

**_Check my comments below :point_down:_** 

- **_The program is not working, please fix it and make it work as expected._**

#### _Reviewed by Temitayo Adeleke._

**Review by christianotieno — APPROVED** [review]
# Project Approved 🍾 
Good job with the changes @carlosprietobarron. The UI looks good too. Keep it up.👏 
All the required changes have been met in this project. I will go ahead and mark this as complete. 👍 
I wish you all the very best in your next tasks 🥇 

##

#### Christian Otieno (TSE)

**Pull request merged**